### PR TITLE
fix(CI): Fix buildx for 1.12.x branch

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: udit <udit@chaosnative.com>

- Update docker buildx command and remove `sudo` in the push script to fix the broken CI.

### CODE REVIEW

**Review by ksatchit — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name: Build
 on:
   pull_request:
-    branches: [master]
+    branches: [dev]
```
Comment (ksatchit) [inline .github/workflows/build.yml:5]: Are we sure of this?

Comment (uditgaurav) [reply .github/workflows/build.yml]: Have updated!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -68,7 +68,7 @@ jobs:
         env:
           DOCKER_REPO: litmuschaos
           DOCKER_IMAGE: go-runner
-          DOCKER_TAG: ci
+          DOCKER_TAG: 1.12.4
```
Comment (ksatchit) [inline .github/workflows/build.yml:71]: Should this be 1.12.x-ci ? With 1.12.4 being pushed on release tag creation?

Comment (uditgaurav) [reply .github/workflows/build.yml:71]: No, right now the content of `v1.12.x` will be pushed with the `1.12.4` tag, for the release tag creation event we have a different workflow that is `release.yml` which will create a new tag.
Having said that for pushing any new changes into this branch we'll need to update the image tag from 1.12.4

Comment (ksatchit) [reply .github/workflows/build.yml]: Docker tags for release should be auto-generated on gh release. With regular pushes to a branch resulting in a <branch> specific ci build.

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**